components/raftstore/src/store/fsm/store.rs

```diff
@@ -1461,11 +1486,52 @@ impl<'a, T: Transport, C: PdClient> StoreFsmDelegate<'a, T, C> {
             // Target tombstone peer doesn't exist, so ignore it.
             return Ok(());
         }
-        if self.check_msg(&msg)? {
-            return Ok(());
-        }
-        if !self.maybe_create_peer(region_id, &msg)? {
-            return Ok(());
+        let check_msg_status = self.check_msg(&msg)?;
+        match check_msg_status {
+            CheckMsgStatus::DropMsg => return Ok(()),
+            CheckMsgStatus::PeerExist => (),
+            CheckMsgStatus::NewPeer | CheckMsgStatus::NewPeerFirst => {
+                if !is_initial_msg(msg.get_message()) {
```

Comment:
This case is very unlikely to happen because we have already checked in `check_msg` and it's wasteful to update `pending_create_peers` and read `RegionLocalState` if it's not an initial msg.